src/dbcsr_api_c.F

```diff
@@ -300,4 +300,377 @@ SUBROUTINE c_dbcsr_multiply_d(transa, transb, alpha, c_matrix_a, c_matrix_b, bet
+
+      LOGICAL                                            :: fa
+      TYPE(dbcsr_type), POINTER                          :: matrix_a, matrix_b
+      CHARACTER                                          :: c_matrix_type
+
+      c_matrix_type = 'N'
+      CALL c_f_pointer(c_matrix_a, matrix_a)
+      CALL c_f_pointer(c_matrix_b, matrix_b)
+      fa = .FALSE.
+      CALL dbcsr_copy(matrix_a, matrix_b,"matrix-copy", fa,fa,fa,c_matrix_type)
+   END SUBROUTINE
+
+! **************************************************************************************************
+!> \brief ...
+!> \param c_matrix_a ...
+!> \param c_matrix_b ...
+!> \param faca ...
+!> \param facb ...
+!> \param C ...
+!> \param name="c_dbcsr_add_d" ...
+! **************************************************************************************************
+   SUBROUTINE c_dbcsr_add_d(c_matrix_a, c_matrix_b, faca, facb) &
+      bind(C, name="c_dbcsr_add_d")
+      TYPE(c_ptr), INTENT(in)                            :: c_matrix_b
+      TYPE(c_ptr), INTENT(inout)                         :: c_matrix_a
+      REAL(kind=c_double), INTENT(in), value             :: faca,facb
+
+      TYPE(dbcsr_type), POINTER                          :: matrix_a, matrix_b
+
+      CALL c_f_pointer(c_matrix_a, matrix_a)
+      CALL c_f_pointer(c_matrix_b, matrix_b)
+      CALL dbcsr_add(matrix_a, matrix_b, faca, facb)
+
+   END SUBROUTINE
+
+! **************************************************************************************************
+!> \brief ...
+!> \param fcomm ...
+!> \param C ...
+!> \param name="c_dbcsr_finalize_lib_silent" ...
+! **************************************************************************************************
+   SUBROUTINE c_dbcsr_finalise_lib_silent(fcomm) bind(C, name="c_dbcsr_finalize_lib_aux_silent")
+      INTEGER(kind=c_int), INTENT(in)                    :: fcomm
+
+      CALL dbcsr_finalize_lib(fcomm,-1)
+   END SUBROUTINE
+
+! **************************************************************************************************
+!> \brief ...
+!> \param c_matrix_a ...
+!> \param c_matrix_b ...
+!> \param retval ...
+!> \param C ...
+!> \param name="c_dbcsr_trace_ab_d" ...
+! **************************************************************************************************
+   SUBROUTINE c_dbcsr_trace_ab_d(c_matrix_a, c_matrix_b, retval) &
+      bind(C, name="c_dbcsr_trace_ab_d")
+      TYPE(c_ptr), INTENT(in)                            :: c_matrix_b
+      TYPE(c_ptr), INTENT(in)                            :: c_matrix_a
+
+      TYPE(dbcsr_type), POINTER                          :: matrix_a, matrix_b
+      !REAL(kind=c_double), INTENT(inout), value            :: retval
+      REAL(kind=c_double), INTENT(inout)                   :: retval
+
+      CALL c_f_pointer(c_matrix_a, matrix_a)
+      CALL c_f_pointer(c_matrix_b, matrix_b)
+      retval = 0.0
+      call dbcsr_trace(matrix_a, matrix_b, retval)
```

Comment:
Please maintain a consistent style wrt mixing all-upper and lower-case, and further down wrt the indentation. Running `make pretty` can help with that.